### PR TITLE
Fixing opensearch helmchart conditions when there is only cluster_manager role specified to nodes

### PR DESCRIPTION
**Pull request by MindAwakeBodyAsleep**
### Description
As new opensearch role for nodes was introduce helmchart templates should be adjusted. I Fixed opensearch helmchart conditions when there is no master role specified to nodes. 
### Issues Resolved
[List any issues this PR will resolve. You should likely open an issue if one does not already exist.]
 
### Check List
- [x] Commits are signed per the DCO using --signoff

For any changes to files within Helm chart directories:
- [x] Helm chart version bumped
- [x] Helm chart `CHANGELOG.md` updated to reflect change

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/helm-charts/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by MindAwakeBodyAsleep** [comment]
@TheAlgo @DandyDeveloper have you had a chance to take a look on that PR? :)

**Comment by bbarani** [comment]
@TheAlgo @DandyDeveloper @peterzhuamazon @prudhvigodithi Can you please review this PR?

**Comment by prudhvigodithi** [comment]
Hey @MindAwakeBodyAsleep thanks for the contribution, the condition added LGTM, but can you please share the issue you are facing ?
Thank you

File: charts/opensearch/templates/statefulset.yaml [hunk 1]
```diff
@@ -354,7 +354,7 @@ spec:
           valueFrom:
             fieldRef:
               fieldPath: metadata.name
-        {{- if (and (has "master" .Values.roles) (not .Values.singleNode)) }}
+        {{- if (and (or (has "master" .Values.roles) (has "cluster_manager" .Values.roles)) (not .Values.singleNode)) }}
```
Comment (TheAlgo) [inline charts/opensearch/templates/statefulset.yaml:357]: Good to add description in README.md as well for the same. Also echoing on the question by @prudhvigodithi how is it impacting and what are the failures we are seeing. Good to capture it in the github issue

**Review by TheAlgo — CHANGES_REQUESTED** [review]

**Comment by peterzhuamazon** [comment]
This seems only a change for 2.x+ as 1.x does not have cluster_manager term.
So no backport.
@prudhvigodithi

**Comment by prudhvigodithi** [comment]
Hey @MindAwakeBodyAsleep just following back, can you please take care of the previous comments and we can merge this PR.
Thank you

**Comment by prudhvigodithi** [comment]
Hey @MindAwakeBodyAsleep just following back, can you please take care of the previous comments and we can merge this PR, also I do see some conflicts can you please fix them as well?
Thank you

**Review by TheAlgo — CHANGES_REQUESTED** [review]
@MindAwakeBodyAsleep Checking for the comments on the PR. Thanks

**Comment by TheAlgo** [comment]
@MindAwakeBodyAsleep Checking for the comments on the PR. Thanks

File: charts/opensearch/templates/statefulset.yaml (hunk 1 above)
Comment (Phenix66) [reply charts/opensearch/templates/statefulset.yaml:357]: I can provide some input on the question posed by @prudhvigodithi . I recently was deploying Opensearch for the first time and was having issues when there was more than one replica. I had been attempting to use all inclusive language, to include specifying a new `roles` key in my `values.yaml` where I replaced the `master` role with `cluster_manger`. I tracked down the issue to these conditional renderings in the helm chart. I solved it by just using the non-inclusive language.

**Comment by dblock** [comment]
Looks stalled, closing, please feel free to finish and reopen.

[Catch All Triage - [1](https://github.com/dblock), [2](https://github.com/reta), [3](https://github.com/krisfreedain), [4](https://github.com/OVI3D0)]

**Comment by Phenix66** [comment]
Glad I received the email notification on this, I had completely forgotten about this and wanted to keep tabs on it. Since the original author didn't seem to follow through, I created a new PR that covers all inclusive language changes: https://github.com/opensearch-project/helm-charts/pull/560

**Comment by prudhvigodithi** [comment]
Thanks @Phenix66.

**Pull request closed without merging**